pkg/ebpf/probes/probes.go

```diff
@@ -2,178 +2,503 @@ package probes
-			SecurityInodeSymlink:       {event: "security_inode_symlink", probeType: kprobe, fn: "trace_security_inode_symlink"},
-			SecurityInodeUnlink:        {event: "security_inode_unlink", probeType: kprobe, fn: "trace_security_inode_unlink"},
-			SecurityMmapAddr:           {event: "security_mmap_addr", probeType: kprobe, fn: "trace_mmap_alert"},
-			SecurityMmapFile:           {event: "security_mmap_file", probeType: kprobe, fn: "trace_security_mmap_file"},
-			DoSplice:                   {event: "do_splice", probeType: kprobe, fn: "trace_do_splice"},
-			DoSpliceRet:                {event: "do_splice", probeType: kretprobe, fn: "trace_ret_do_splice"},
-			ProcCreate:                 {event: "proc_create", probeType: kprobe, fn: "trace_proc_create"},
-			SecurityFileMProtect:       {event: "security_file_mprotect", probeType: kprobe, fn: "trace_security_file_mprotect"},
-			CommitCreds:                {event: "commit_creds", probeType: kprobe, fn: "trace_commit_creds"},
-			SwitchTaskNS:               {event: "switch_task_namespaces", probeType: kprobe, fn: "trace_switch_task_namespaces"},
-			ARMKprobe:                  {event: "arm_kprobe", probeType: kprobe, fn: "trace_arm_kprobe"},
-			CallUsermodeHelper:         {event: "call_usermodehelper", probeType: kprobe, fn: "trace_call_usermodehelper"},
-			DebugfsCreateFile:          {event: "debugfs_create_file", probeType: kprobe, fn: "trace_debugfs_create_file"},
-			DebugfsCreateDir:           {event: "debugfs_create_dir", probeType: kprobe, fn: "trace_debugfs_create_dir"},
-			DeviceAdd:                  {event: "device_add", probeType: kprobe, fn: "trace_device_add"},
-			RegisterChrdev:             {event: "__register_chrdev", probeType: kprobe, fn: "trace___register_chrdev"},
-			RegisterChrdevRet:          {event: "__register_chrdev", probeType: kretprobe, fn: "trace_ret__register_chrdev"},
-			DoInitModule:               {event: "do_init_module", probeType: kprobe, fn: "trace_do_init_module"},
-			DoInitModuleRet:            {event: "do_init_module", probeType: kretprobe, fn: "trace_ret_do_init_module"},
-			LoadElfPhdrs:               {event: "load_elf_phdrs", probeType: kprobe, fn: "trace_load_elf_phdrs"},
-			Filldir64:                  {event: "filldir64", probeType: kprobe, fn: "trace_filldir64"},
-			TaskRename:                 {event: "task:task_rename", probeType: rawTracepoint, fn: "tracepoint__task__task_rename"},
-		},
+		probes: allProbes,
 		module: module,
 	}, nil
 }
 
-//Attach the probe corresponding to the probe's handle
-func (p *probes) Attach(probe Handle) error {
-	if p.probes[probe] == nil {
-		return fmt.Errorf("failed to attach probe: doesn't exist")
+func (p *probes) Attach(args ...interface{}) error {
+	return p.attachOrDetach(true, args...)
+}
+
+func (p *probes) Detach(args ...interface{}) error {
+	return p.attachOrDetach(false, args...)
+}
+
+func (p *probes) attachOrDetach(attach bool, args ...interface{}) error {
```

Comment:
There was a discussion between Nadav and I about variadic functions being used here but I used it on purpose so the API could accept any argument (depending on existing and future probe types we may have).

I explained how to use the API here: https://github.com/aquasecurity/tracee/pull/1820/files#diff-a4767f3b581014c47b886c59c79fcf8374aee0edece50c13d6e9758df78abf53R12 and, despite preferences on implementation, the caller won't be affected by this decision.

For now user will attach a probe or detach it. User will also attach a tcProbe to an interface or detach it (and the API supports both, which is good enough and much better than what we currently have).

I'll the one using this interface to implement dynamic interface attachment/detachment and, if it does not serve its purpose, I could rethink it but, for now, I think it is good.